src/garage/replay_buffer/replay_buffer.py

```diff
@@ -12,173 +12,75 @@
 import abc
 from abc import abstractmethod
 
-import numpy as np
-
 
 class ReplayBuffer(metaclass=abc.ABCMeta):
     """Abstract class for Replay Buffer.
 
     Args:
         env_spec (garage.envs.EnvSpec): Environment specification.
-        size_in_transitions (int): total size of transitions in the buffer
-        time_horizon (int): time horizon of rollout.
-
+        capacity_in_transitions (int): capacity of the replay buffer,
+            measured in transitions.
     """
 
-    def __init__(self, env_spec, size_in_transitions, time_horizon):
-        del env_spec
+    def __init__(self, env_spec, capacity_in_transitions):
```

Comment:
I think that interfaces should be constructor-less and have no default implementations. If there's an obvious typical implementation that can be a simple abstract class inherited from the interface.

For instance, there's no reason a replay buffer doesn't need to know the capacity up front, in which case forcing this constructor args is just a useless burden.

Reply:
My personal opinion on this topic is very simple: `__init__` should not be part of an API, ever. When I'm constructing something, I know exactly what type it is, so the API doesn't mean anything. The only reason to have `__init__` be part of an API is that you want to have default method implementations in terms of private fields, which I think is an anti-pattern.

We have some places where we've done this already in garage, but it's resulted in difficult to explain or fix bugs (see torch policies / issue #1141). The problem is that implementing multiple interfaces on the same type (which is very common in python, once you start looking) then requires reasoning about multiple inheritance, which will always be somewhat broken.